airflow/providers/docker/operators/docker.py

```diff
@@ -333,7 +333,7 @@ def _run_image_with_mounts(
             return None
         finally:
             if self.auto_remove:
-                self.cli.remove_container(self.container['Id'])
+                self.cli.remove_container(self.container['Id'], force=True)
```

Comment:
I can imagine someone would want to debug a container if it fails.

Maybe we should change `auto_remove` to accept either boolean (for compatibility) or a str

```python
DockerOperator(
    ...,
    # auto_remove="never",  # Default
    # auto_remove="success",  # Same as 'True'
    # auto_remove="force",  # Add force=True
)

Reply:
@uranusjr Thanks for the review, there may be misunderstandings here,

Imagine two scenarios:
1. The dagrun is healthy, the container failed.
2.dagrun failed, but the container is healthy and still running.

I think What you described above is Scenario 1, what I came across is Scenario 2.

For scenario 1, the current code is able to remove the container because the container is failed.

But in scenario 2, the container will not be remove because the container is still running.

For scenario 2, I think there is no need to debug the container, because the container is healthy.

Reply:
No, I was talking about the second scenario. When a run fails but the container is still running, it would be possible for the user to attach into the container to see what failed. But if we do `-f`, that would no longer be possible.

Reply:
But a running container will not cause dagrun to failed.

If the dagrun failed and the container is still running, the dagrun is likely to failed due to factors outside the container. 

In this case, we should debug the reasons outside the container instead of inside the container?

Reply:
There are a lot of things that can fail, including programming errors or edge cases not handled by the task code. And in those cases, having the exact environment that caused the unanticipated failure makes things easier to debug, since it contains as much context as possible.

Reply:
Ok ,I have updated the code, please review